### ISSUE TITLE
SC2164

### ISSUE THREAD

**Issue by zefir-git** [koalaman/shellcheck#3003]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2164
- My shellcheck version (`shellcheck --version` or "online"): online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/zsh
function foo() {
    cd generated_files
    return $?
}

```

#### Here's what shellcheck currently says:
SC2164: suggested `cd generated_files || exit`

#### Here's what I wanted or expected to see:
No suggestion. The function returns on the next line anyways.

**Comment by brother** [comment]
[The warning is correct](https://www.shellcheck.net/wiki/SC2164). [Ignore it with a directive](https://www.shellcheck.net/wiki/Directive) if you don't like it.

**Comment by zefir-git** [comment]
If `cd` fails, it will return on the next line anyways.

this doesn't make sense:
```sh
cd generated_files || return $?
return $?
```

Isn't this like?
```
int result = cd("generated_files");
if (result == 1) return result;
return result;
```

`|| return` when the next line already is return seems redundant to me.
